File: GymLogServer/GymLogServer/DTOs/TrainsDTO.cs
using System.ComponentModel.DataAnnotations;

namespace GymLogServer.DTOs
{
    public class TrainsDTO
    {
        [Required(ErrorMessage = "Type must be")]
        public string Type { get; set; }
        public string Description { get; set; }
        public DateTime Date { get; set; }
        [Range(1, int.MaxValue, ErrorMessage = "Duration must be greater than 0")]
        public int Duration { get; set; }
        public int UserId { get; set; }
    }
}

File: GymLogApp/app/src/main/java/com/example/gymlogapp/model/TrainsDTO.kt
package com.example.gymlogapp.model

data class TrainsDTO(
    val type: String,
    val description: String? = null,
    val date: String,          // "yyyy-MM-dd" или "yyyy-MM-ddTHH:mm:ss"
    val duration: Int,
    val userId: Int = 0        // сервер берёт из токена, так что можно 0
)